### ISSUE TITLE
User guide for new users

### ISSUE THREAD

**Issue by docwilmot** [backdrop-ops/backdropcms.org#55]
Moving from https://github.com/backdrop/backdrop-issues/issues/808

We need to create a well-thought and organised user guide for the site owner / site editor / site admin type. Current proposal is to create a new section of backdropcms.org for this purpose. The section should be organized however is needed to make it clear and understandable.

This new section will be linked from the Handbook, in the user guide area.

**Google docs**: https://drive.google.com/folderview?id=0B5vNMqFUUd-SfjV0Z1hZUUlrc2N5anRwM2lDNm0zOUlRVThkZ3VEZ1I1UXJnT3N5dG10TjA&usp=sharing
- [x] Quick start guide
- [x] Modules themes and layouts
- [x] Nodes and Entities
- [x] Understanding Layouts
- [x] Blocks
  - [x] Configuring blocks: examples
- [x] Content types
  - [x] Create a custom content type
  - [x] Add fields to a content type
- [x] Display settings (View modes)
- [x] The Admin bar
- [x] Menus
- [x] Project installer
- [x] Taxonomy and categorizing content
- [x] Users, roles and permissions
- [x] Views basics
  - [x] Views exercises
  - [x] Configuring Views - advanced
  - [x] Configuring Views
- [ ] Installation
- [ ] Themes
  - [ ] Color
- [ ] Creating content
- [ ] Multilingual
- [ ] Configuration Management
- [ ] Multisite
- [ ] Book
- [ ] Ckeditor
- [ ] Comment
- [ ] Contact forms
- [ ] Site reports
- [ ] Fields
  - [ ] Date
  - [ ] File
  - [ ] Image
- [ ] Filters
- [ ] Updating project code
- [ ] Pathauto
- [ ] Search
- [ ] Redirects
- [ ] Howtos
  - [ ] Customizing the front page
  - [ ] Creating lists of content
  - [ ] Create an image gallery
  - [ ] Customize the User Registration Form
  - [ ] Create a Blog site
  - [ ] Create a personal site
  - [ ] Members-only site

**Comment by Graham-72** [comment]
This is an excellent list and if we can produce a guide like this it will be very useful to someone setting out to build a site using Backdrop.

What I have in mind as a guide for site editors would be very much a subset of this. My users (for whom I have built sites) would run a mile, probably to Wordpress (!) if I gave them this to read.

I have just caught up with this blog - http://www.istos.it/en/blog/learning-drupal/how-i-finally-convinced-my-dad-switch-wordpress-drupal - which sounded very familiar to me, though of course things have moved on since that was written, and Backdrop is a great deal more user-friendly than Drupal 7. (Thanks guys!)

Can we pick out the parts for the non-techie user?

**Comment by docwilmot** [comment]
@Graham-72 Can we have a similar outline of what you need for the non-techs among us? I see no reason why we cant have layers of documentation as you say.

**Comment by quicksketch** [comment]
> - Code snippets

**Comment by docwilmot** [comment]
> Code "snippets" are generally a dangerous thing

I understand "code snippets" to mean howto samples, like: 
```
//how to get a list of all layouts in PHP:
$all_layouts = YOUR_MODULE_layout_load_layouts() // Made-up code
```

NOT the "put this in your node content" type of stuff. Snippets were very helpful for me in figuring out what to do when...
Also I would now like to suggest that rather than allowing free editing and contributing of documentation (all of them, this, API, etc) we have an approval process.

> general design advice

Meaning, use proper white space, use appropriate fonts, dont use grey text on  greyer backgrounds etc. *design* advice. But I suppose that may either be out of the scope of this effort, or not a priority. Would be a nice addition for an absolute newbie, to prevent some of the OMG websites out there.

> each of the sub-topics don't seem like they'd be worthy of their own page

OK.

**Comment by Graham-72** [comment]
My first thoughts on the content of a basic guide:

Using Backdrop CMS
An introductory guide for site editors

1 Outline of what Backdrop CMS does
1.1 Appearance of your website (themes and styles)
1.2 User permissions – who can do what
1.3 What makes a page (concept of nodes,  layouts, blocks, navigation)
1.4 Revisions and backups
1.5 Roles of site designer and administrator

2 Controlling content
2.1 Editing text
2.2 Adding images
2.3 Uploading documents
2.5 Links and anchors
2.6 Content types
2.7 Page layout
2.8 The Home Page

3 Navigation
3.1 Main menu
3.2 Secondary menu
3.3 User-friendly URLs
3.4 Exclude from menu

4 Comments
4.1 Allowing comments
4.2 User registration
4.3 Removing comments

5 Basic administration
5.1 Admin menu
5.2 Finding content
5.3 File folders
5.4 Reports
5.5 Software updates

6 Opportunities
6.1 Changing the appearance
6.2 Adding functionality (contributed modules)
6.3 Customised views
6.4 Linking social media
6.5 Other resources
6.6 Getting help

**Comment by docwilmot** [comment]
@Graham-72 your list seems more an intersection than a subset, dont you think? How about if the guide was as comprehensive as my list but we tagged each article by who it could apply to (owner, admin, editor etc)? And/or tag by complexity (basic > advanced > guru > quicksketch) ?

**Comment by Graham-72** [comment]
@docwilmot yes I agree my list hasn't turned out as a subset. I think what I have in mind is a booklet that I would hand to a new user as an introduction, and would avoid including extra layers of complexity that some might get interested in later i.e. the guide you have outlined. When I say booklet I do really mean something that could be printed on paper - horrors!

Maybe I should go ahead and write something that might interest a publisher ? :-) [How does one add smileys in GitHub?]

**Comment by quicksketch** [comment]
> [How does one add smileys in GitHub?]

`:smile:` :smile:

**Comment by jenlampton** [comment]
+1 to all the wonderful work going on in this issue!

**Comment by cellear** [comment]
I teach an introductory web development class at Berkeley.  On the last class of my last session, I had my students download and spin up Backdrop on Pantheon to see what their experience was.  They were able to get it installed and running just fine, but when the installation process finished they had no idea what to do next.  It was a little painful for me to watch them click randomly on menu items that don't have any effect until you have some content.

So...why don't we give them a subset of @Graham-72 / @docwilmot 's outline as a single default node?  Just the TOC/outline, linked to online text.  That way beginners would have something to dig into, and experienced people could just delete or re-purpose the node.

**Comment by docwilmot** [comment]
[Or guided tours?](https://github.com/backdrop-contrib/joyride)

**Comment by jenlampton** [comment]
@cellear because on 99% of Drupal sites that node would be deleted.

I'm all in favor of improving the first-time experience, but I don't think filling the site with trash content is the way to do it.  An overview page that's not a node - maybe.  But certainly not a node.

@docwilmot for the issue of tour module in core, see https://github.com/backdrop/backdrop-issues/issues/89

**Comment by docwilmot** [comment]
> let's see if a tour module develops in contrib first

It did. That was the link above.

**Comment by cellear** [comment]
Ouch.   Trash content? 

It seems to work ok for WordPress.

**Comment by jenlampton** [comment]
trash content as in "99% of people will throw this in the trash" not as in the documentation itself is trash ;)  maybe about-to-be-trashed content?

**Comment by cellear** [comment]
But that's the point!   It seems to work ok for WordPress.   When you start a new WP site,  it creates a node that says this:
>This is an example of a page. Unlike posts, which are displayed on your blog’s front page in the order they’re published, pages are better suited for more timeless content that you want to be easily accessible, like your About or Contact information. Click the Edit link to make changes to this page or add another page.

I'm proposing we do something similar,  though I figured as long as we were making a page,  it might as well point to something useful. The WP example page,  while better than Drupal's Nothing,  assumes a familiarity with jargon that it probably shouldn't. (It assumes the new user is making a blog,  which seems to unnecessarily box them in.)

So yes,  make things easier for beginners,  even if it doesn't help the gurus.  Isn't that a core Backdrop principle?

**Comment by cellear** [comment]
(I'd like to cross-reference this: https://github.com/backdrop/backdrop-issues/issues/519 but I'm not sure how to.)

**Comment by Graham-72** [comment]
I like the idea of the tour module though, as with videos, I like to be able to navigate (either forwards or backwards) to revisit particular points in an explanation. Is this possible?

I have made a start at writing text for the basic guide I envisage. I have produced a first part as a PDF but GitHub doesn't seem to accept these, so I have put it on my webserver at http://www.oliverweb.net/BackdropEditingGuide_Draft.pdf

I must emphasise that I am writing this for the non-technical user who is suddenly challenged with the task of editing a website, either for themself or for a small organisation - a situation that I have met on a few occasions with Drupal sites. Of course we also need a clear technical guide that makes it as easy as possible for someone to set up and run a website using Backdrop.

**Comment by jenlampton** [comment]
> This is an example of a page. Unlike posts, which are displayed on your blog’s front page in the order they’re published, pages are better suited for more timeless content that you want to be easily accessible, like your About or Contact information. Click the Edit link to make changes to this page or add another page.

@cellear But, wouldn't this information be useful to a second or third person, even after this first piece of content has been updated, and the information removed? My point is that if we are going to provide something of value - it shouldn't be in a node, or somewhere where it is *intended* to be deleted.  

WordPress assumes that the person building the site will be the person creating / editing content. So the message in a default node is likely to be seen by the 'correct' person. In Drupal and Backdrop the person building the site is the person least likely to need instructions on how to edit content, and the person most likely to delete or edit away the valuable information before it gets to the eyes of those who need it.

Yes, we need to make things easier for beginners (and the first time experience is important!). But hopefully we can make things easier for all site users, not just the person (more likely the guru) who built the site.

**Comment by Graham-72** [comment]
> In Drupal and Backdrop the person building the site is the person least likely to need instructions on how to edit content

@jenlampton  My thoughts exactly!

**Comment by docwilmot** [comment]
> I like to be able to navigate (either forwards or backwards) to revisit particular points in an explanation.

This simple sentence has cost me a good week. I checked, and found that Joyride version used in D8 doesnt do backward navigation; this feature requires either [a patch](https://github.com/zurb/joyride/pull/84) or updating to the latest Joyride.

BUT the previously standalone Joyride plugin has now been integrated into Zurb's Foundation, and to use the latest code requires some more hacking and/or including the full Foundation js into the tour module. Thats tough.

So a few Google hours of search found ["Hopscotch" by Linkedin](http://linkedin.github.io/hopscotch/) which has the following advantages:

- Actively updated, small plugin
- Includes Back and Forward navigation
- Multiple tours per page would be possible (not doable with Joyride)
- Tours spanning multiple pages also possible (requires some hacks again for Joyride)
- Nicer looking API (I'm nowhere near an expert, but this seemed much easier to deal with than Joyride)

So, therefore, I've adopted the existing Joyride module to use Hopscotch. See https://github.com/docwilmot/hopscotch. Still rough, please review and test.

My aim now, would be to build tours in tandem with the docs. 

Comments welcome.

**Comment by docwilmot** [comment]
Going to go back to writing docs again, but will start with the developer docs as thats already on the way. 

@Graham-72 I've made a folder on Google Docs [here](https://drive.google.com/folderview?id=0B5vNMqFUUd-SflU4M25jb3VaTXZDVHJlUldubGtFcnJsb3J0RHBTM3BqT0FFTlhGT1N2alE&usp=sharing). Maybe we could start the user docs with what youve already got, if you dont mind? 

I've mad a checklist out of the first post; create a new Google doc for one of the checklist items and go ahead.

**Comment by Graham-72** [comment]
@docwilmot thanks for creating the folder on Google Docs - excellent idea. I have put the current version of my embryo User Guide in there and will shortly resume work on it. I am intending to use the new admin styles for screenshots. Also, use revised terminology for 'article' and 'basic page'.

**Comment by Graham-72** [comment]
@docwilmot I wonder whether we could have a third subfolder in Google docs for useful notes and tips for Backdrop developers? For example, I am gradually learning how to do different things with PhpStorm and GitHub. I have to write notes for myself in order not to forget things and these might be useful if collected together. Also various tips have been written by you, Nate and others in these issues, and it could be helpful to record them in one place. What do you think?

**Comment by docwilmot** [comment]
Where would we put user docs?

**Comment by Graham-72** [comment]
On backdropcms.org?

**Comment by docwilmot** [comment]
Figured, but it would need building and styling.

**Comment by Graham-72** [comment]
Not just a downloadable PDF?

**Comment by docwilmot** [comment]
No, has to be searchable, categorized, version-filtered help.

**Comment by Graham-72** [comment]
If you say so; you know best.

**Comment by docwilmot** [comment]
I dont actually; I write these things out so the important people can see what I'm thinking. Admittedly, I'm rarely sure what I'm doing here.

**Comment by Graham-72** [comment]
Well, you certainly know better than I do, but let's see what they say.
I know how to produce a PDF, but not much clue about the other :smile:

**Comment by klonos** [comment]
We can have the best of both worlds: [Printer, email and PDF versions](https://www.drupal.org/project/print)

**Comment by docwilmot** [comment]
Just a reminder: we need space on B.org for this. Just a content type at least, and we can start writing them unpublished? @jenlampton you can then start publishing them afterwards once the display page is themed and built and prettified.

**Comment by jenlampton** [comment]
I didn't have time to re-read the whole strand of comments, but I think the outline as it is now (in the top issue) is way too much information. People are easily deterred by thinking there is too much to learn, and providing an outline that rich, though intended to help, is just as likely to turn people away.

In addition to (possibly) having the opposite effect as intended, it's also going to be very hard to keep something this comprehensive up to date. We are constantly adding new things into core, and sometimes even moving the locations of existing things to improve usability. If we cover all the items on this list, we'll be spending just as much time updating the documentation as we are updating the code. Drupal.org solved this problem by allowing everyone to help maintain documentation, and the handbook grew and grew. If possible I would like to avoid that same outcome by controlling the scope here more closely.

That said, I would love to have a user guide on backdropcms.org, and want to move this forward. @docwilmot I'm going to stub out a few pages in the existing handbook so you can get started with some of these concepts.

**Comment by jenlampton** [comment]
I've updated the parent item with a minimal sub-set of items. Let's start here, and work our way up :)

**Comment by docwilmot** [comment]
@jenlampton I am a bit disappointed with the approach to this. The outline was created after some research into the user docs on Drupal and WP and some thought went into it. Furthermore I think a couple of us had already agreed that it would be beneficial to have this level of documentation. A decision to review the size of it would be reasonable, but a unilateral decision to delete it and replace it seems inappropriate.

Whats more, I disagree with you, to make things worse. 1) I haven't come across any research to suggest  "People are easily deterred by thinking there is too much to learn"; 2) the outline would not be visible to users, only the top-level menu items, so users would not be seeing the depth of information unless they actually wanted to find it; 3) the list is **tiny** compared to Drupal and WP. This is WP user docs front page:
![wphelp1](https://cloud.githubusercontent.com/assets/1042283/12451404/351d9ae2-bf57-11e5-8fe5-f7aa04d2d881.PNG)
And this is **one** subpage from linked from that front page:
![wphelp2](https://cloud.githubusercontent.com/assets/1042283/12451418/4bf2344e-bf57-11e5-8387-827f89f97a43.PNG)

Finally, the plan was for **user** docs. What you've proposed, as judged by the proposed titles,  is already reflected in our API/developer docs. THese were docs for people who may not have used Drupal/Backdrop before and we recommended a section for absolute beginners, and a gradual progression of the depth of the documentation.

In summary though, I would prefer if we had a bit more consultation before decisions like this, which veto previous agreements, are made.

**Comment by Graham-72** [comment]
I am back with the strong feeling that we must address our users in two categories, both very important:

1. Those who are setting out to build a website with Backdrop.

2. Those who have been asked to edit content in a Backdrop website.

I do (1) myself and can cope with masses of information, though a coherent guide will be very welcome, and essential for promoting Backdrop as an easy CMS for web designers to use.

I pass my work on to users in category (2) and usually they are initially almost terrified of making an alteration to a page that can be instantly seen from around the world. Such users are definitely deterred by too much information. Please, please can we have a cut-down 'site editors guide' for these people. I did draft something now available again at  http://www.oliverweb.net/BackdropEditingGuide_Draft.pdf and I am about to resurrect it in order to use part of it to explain to a group of users new to Backdrop how they can create an email to send (via views_send) to their club members.

And yes, there are other categories of user, such as the person who has some admin responsibility for the site.

**Comment by jenlampton** [comment]
Sorry @docwilmot, I wasn't attempting to make a unilateral decision - just to start with a lowest common denominator that we could all agree on and build up from there. My intent was to unblock this issue so work could proceed even though the outline wasn't finalized.

Perhaps you are right that we should start with something well-thought through and stick to that. 

I think there may be some confusion as to what (and who?) this documentation is for.  If we can clear that up then maybe we'll be able to decide how much there should be and where it should live.

> What you've proposed, as judged by the proposed titles, is already reflected in our API/developer docs.

To be clear, the information I'd proposed site is NOT intended for developers. Information for developers should remain on the API site, and has no business on backdropcms.org. I'd wanted this information to be suitable for complete beginners: an overview of basic but important concepts. NO CODE.

I think another reason I had been hoping to keep these pages (really really) short, is because I'd placed them into the handbook in the `User Guide` section, since that section is a little thin and needs to be fleshed out.  However, the handbook needs to be short - someone needs to be able to read all of it in a short amount of time. The handbook is not the place for recipes on how to build a site or details on how to add contextual filters to a view.

To address @Graham-72's point, I think we do need two things. Perhaps the `User guide` should be this (very) short sweet overview of concepts, but we also need more thorough documentation (that' not in the handbook) for Administrators or people who will actually be building websites.

How would you feel about that?

**Comment by docwilmot** [comment]
No worries @jenlampton 
The thought was to have a section labelled clearly for beginners, but still have it in as part of the general docs, as for example in the image of WP docs in my last post. ie "Learn How to Use Backdrop" or "Backdrop for the Absolute Beginner". I suspect that might be easier long term than maintaining two separate sets (three is you count the API site). We could highlight/theme that section to make it easily discoverable by newbies in some way.
I think it would probably be better to have a link in the handbook to a proper docs collection rather than try to build the docs in that little drawer.
And it seems at least for @Graham-72 that the absolute beginners would be a priority, and he's already built a great start, so if we could agree on the topics for this, we could begin with his PDF?

**Comment by klonos** [comment]
I doesn't have to be different section per se (apart of the API site of course that needs to be on its own, separately from the user guides).

I was thinking about a single documentation site that would branch in 3 (to begin with) main sections...

1. "Build your *first* website with Backdrop"
  Notice the word "first" there that denotes the intended audience which will be **beginners**. This section would include basic steps like walk through installation, choose site name/slogan, create a page or two, create a contact form etc. ...perhaps also choose one of the core-supplied themes (and to change colors if the theme supports it) or guide how to install and enable a contrib one but that's as far as we go with themes in this section.

2. "Extend your website with *advanced* features"
  Again notice the word "advanced" denoting intended audience which will mainly be **site builders and admins**. This would include two sub-sections: one for configuring core features and one for features that are offered by contrib modules.

3. "Customize the look of your website"
  This would delve into layouts, themes as well as creating custom themes and subthemes (say from a given .PSD template). Intended of course for **web designers and themers***.

I believe that these 3 categories should be enough for starters. We can add, remove or move things around as we see fit with time. For example a 4th section on how to code and create contrib modules.

**Comment by docwilmot** [comment]
@klonos I like it. I propose, to move forward:
- @Graham-72 has already a good start on the "Build your first website with Backdrop", and we should use this as a template
- @Graham-72 upload the text version of his PDF user guide somewhere editable so we can start making it ready. We can use my [Google Docs link](https://drive.google.com/folderview?id=0B5vNMqFUUd-SflU4M25jb3VaTXZDVHJlUldubGtFcnJsb3J0RHBTM3BqT0FFTlhGT1N2alE&usp=sharing) still.
- Once that's on the way, we will need a homepage on Backdrop.org that is titled and is styled and designed for docs.
- The handbook drawer should have a link to the docs homepage, and not try to hold the actual docs.
- Then later or concurrently, depending on resources, we work on the other two sections

**Comment by klonos** [comment]
Sounds like a plan. @jenlampton, @quicksketch once the documentation site/section is ready, can I please have edit access for the documentation nodes?

**Comment by klonos** [comment]
...and can we please have revisions enabled on these nodes? It would help to present diffs between edits as requests for comments/approval (until we have a proper publishing workflow in place).

**Comment by Graham-72** [comment]
'upload the text version'
Please advise me on the best format to use.   Should it be plain txt, perhaps with paragraph numbering for easy reference,  or should it be .md, or  something else?  I will gladly do it.

**Comment by docwilmot** [comment]
Just pain text, no need for numbering. Simple section headers would be fine. It'll be easier to do any formatting and numbering once we're happy with the actual content. 

We could then save each proposed book page to a separate text file (cut and paste) or just work on the one while text file, depending on the size. 

Anyone of you proficient at creating annotated screenshots? And gifs? To make it easier, we should have a chief of graphics and the rest of us just edit. Videos can come later.

**Comment by klonos** [comment]
I can do screenies and gifs. Just let me know ahead because my time is very limited.

**Comment by Graham-72** [comment]
I have now created a repo with the text from my pdf (originally a Word file) at https://github.com/Graham-72/backdrop_guides. Where should I move it if we are all going to be able to edit it?

You will see in the repo that I have separated out my proposed contents list.

Also please note this was written in April 2015 and things have of course changed. I intend to update it soon.

**Comment by jenlampton** [comment]
Unfortunately there isn't a great way to collaborate on documentation.  Github is good with code, but bad with words :(  How about I set up a new content type on backdropcms.org with revisions enabled, and let you guys go wild over there?

**Comment by klonos** [comment]
> How about I set up a new content type on backdropcms.org with revisions enabled, and let you guys go wild over there?

Yes please!

**Comment by jenlampton** [comment]
okay - I've set up a docs content type with revisions. I've made a new 'User guide' menu and you guys should have access to that too, in a minute :)

**Comment by jenlampton** [comment]
ported https://github.com/backdrop-contrib/menu_admin_per_menu. woot! You should now have access to the User guide menu.  Feel free to create & organize whatever you wish in there. Once you are happy with what's there we can add it into the main nav and then we'll get more eyes on it :)

**Comment by Graham-72** [comment]
Excellent. I have posted there my suggested text (from April 2015) and content list.

**Comment by klonos** [comment]
:clap: to both @jenlampton and @Graham-72 

Favorite line in the above comments:

> I need a minute to port a module :)

:smile: :ninja: :zap:

**Comment by klonos** [comment]
> I have posted there my suggested text (from April 2015) and content list.

Link please?

@jenlampton what would be the top-level menu item? Is it all to go under the "Handbook" item in the drawer?

**Comment by klonos** [comment]
...we also need to have pathauto enabled and access to edit paths please. I think everything should go under either ```https://backdropcms.org/documentation``` or ```https://backdropcms.org/docs``` for added SEO. ...or just stick with the current `https://backdropcms.org/guide`.

Current links from the "User Guide" section (under "Handbook") end respective paths:

System Requirements | `https://backdropcms.org/requirements`
Installation | `https://backdropcms.org/installation`
Skin with themes | `https://backdropcms.org/guide/themes`
Build with Layouts | `https://backdropcms.org/guide/layouts`
Extend with modules | `https://backdropcms.org/guide/modules`

I believe all these should be under a single path. If we require "quick links" for some of them, then use redirects.

**Comment by klonos** [comment]
...it's just that plain "guide" triggers my OCD.

**Comment by klonos** [comment]
@Graham-72 ...nothing listed when searching for ["Documentation" nodes](https://backdropcms.org/admin/content/node?status=All&type[]=docs). Did you publish that node?

**Comment by Graham-72** [comment]
@klonos It is an unpublished node on backdropcms.org of type 'Documentation' which hopefully you will find if you log in. I haven't yet found a 'log in' button so I go to the usual backdropcms.org/user and then view the list of content - relevant part is shown in this screenshot:
![admin content - backdrop cms 2016-01-31 11-07-02](https://cloud.githubusercontent.com/assets/10559274/12701757/86c1ec42-c80b-11e5-8b1c-ea7ef096bd29.png)

**Comment by klonos** [comment]
> It is an unpublished node on backdropcms.org of type 'Documentation' which hopefully you will find if you log in.

Unfortunately, we (Backdrop as well as D7) only have the "View **own** unpublished content" permission available which serves me no good :disappointed:

> In Drupal 7, users are not allowed to view any unpublished content by default, not even content created by themselves. By enabling 'View own unpublished content', you grant users access to their own unpublished content.

> The only way to allow users access to all unpublished content in Drupal 7 (without using a contributed module), is to enable the permission 'Bypass content access control'.
Use this permission with extreme caution as it has far more consequences than just allowing access to unpublished content. In combination with Organic Groups for example, this permission allows access to private group content regardless of whether the user is a group member.

source: https://www.drupal.org/node/1106606

...I think we had an issue someplace for implementing this.

@jenlampton up to a ninja port of [View Unpublished](https://www.drupal.org/project/view_unpublished)? :wink: 

> I haven't yet found a 'log in' button so I go to the usual backdropcms.org/user

Same here.

**Comment by klonos** [comment]
> ...I think we had an issue someplace for implementing this.

#897

**Comment by Graham-72** [comment]
Oh! I hadn't appreciated that you wouldn't get to see an unpublished document, so I have now published the two of them. It seems a bit odd to be 'publishing' something that needs more work before it is ready to 'publish' to the wider world.

**Comment by jenlampton** [comment]
> what would be the top-level menu item?

That's up for discussion, but I was thinking we add a link to the handbook called "Codex" to match WP, or "More documentation" or something if you don't like Codex. That page would show shows the whole outline, divided by section, similar to https://codex.wordpress.org

> Oh! I hadn't appreciated that you wouldn't get to see an unpublished document...

Yes, that is odd.  Hold on a few minutes and I'll port another module...

**Comment by jenlampton** [comment]
On second thought, I just decided to fix core: https://github.com/backdrop/backdrop/pull/1231 I'll push this up to b.org as soon as I get tests passing.

**Comment by jenlampton** [comment]
Tests are now passing, so I've deployed the patch.  All editors should have access to all unpublished nodes :)

**Comment by docwilmot** [comment]
https://drive.google.com/folderview?id=0B5vNMqFUUd-SfjV0Z1hZUUlrc2N5anRwM2lDNm0zOUlRVThkZ3VEZ1I1UXJnT3N5dG10TjA&usp=sharing

As promised, I'm working on docs this cycle. Started with the above. Actually all the docs linked here are already on BackdropCMS.org (unpublished) but I've realised it is going to be easier to collaborate on editing if they stay on GDocs till theyve been approved somehow.

The `user docs.txt` file is the master plan, but I've started with the simple stuff first (people who may not know Backdrop nor Drupal), then I'm focusing on the 'new' stuff such as Layout, Config and Installer. Currently on Layout.

@Graham-72 I built a `Quick start Guide` off of your `Backdrop CMS User Guide_Draft` before I realised you had it on Backdrop.org already so we'll need to decide which version to keep.

@klonos you had at some point agreed to be the screenshots guy; would be great if you could peruse the docs and insert placeholders for screenshots (add a <filename.png> to the document matching the screenshot file on your computer). I'm thinking placeholders rather than adding them to the docs to make it easier to move the docs to live once theyre ready. But it may be better to actually embed them in the GDocs files to see how they look so we can vett those too.

@jenlampton I know you had thought we shouldn't be too in-depth so as a compromise, I'm suggesting doing the docs in long form with anchor links, so everything about layouts in one document but subsection titles within the doc can be linked externally. That way less files, but comprehensive nonetheless. Of course feel free to let me know if you think I'm overdoing it.

Finally, every and anybody feel free to comment on the docs (I think you can embed comments directly) if you think additions or deletions are in order, and of course feel free to edit the docs directly if simple things like spelling mistakes.

Any thoughts?

**Comment by jenlampton** [comment]
wow this looks fantastic!!! :)  I'm happy to defer to you and the docs team since it's obvious that you've all be putting a lot of thought and work into this.  +1 for keeping the docs log-form with anchors to jump to a particular part of the page. 

In my experience creating written documentation I found it much easier to place the screenshots in-context for reviewing, but having a image/file naming convention for the screenshots also really helped insert them into the right positions again later.

@docwilmot I'm going to email you some of those documents to you can see if any of the formatting might translate to online docs for Backdrop.

**Comment by klonos** [comment]
> @klonos you had at some point agreed to be the screenshots guy;

I can be that guy :smile: I will be adding them as I have time to be going through the docs. Would be great if people "ping" me or assign documentation issues to me.

@jenlampton, "status - needs screenshots" tag for the world pretty please?

**Comment by klonos** [comment]
...also would be great to either have dedicated "docs - need review", "docs - need work" and "docs - RTBC" tags or rename the current respective "pr - *" tags to remove the "pr - " part. That would generalize them and allow us to use them for documentation issues.

**Comment by jenlampton** [comment]
I really only prefixed them for sorting. Other ideas of a prefix that's
less specific?

- - -
Jennifer Lea Lampton | twitter.com/jenlampton | 510-747-8230
Backdrop CMS & Drupal @ Jeneration Web Development
Check out my new website! | www.jeneration.com

**Comment by Graham-72** [comment]
@docwilmot 
> so we'll need to decide which version to keep

I'm happy that we move forward with your version. 

Is it OK if I draft a new section? I am thinking we should include something about custom node types and the use of Views. My reason is that some sites I have built and handed over to clients contain new content types specifically for that client e.g. data about books that the client adds to (one node per book) and the site displays using views.

Not that I am expecting the user to modify the view; rather I feel they need to know that some pages may be generated automatically and they can add to and edit the data but not the page.

Also, it is a powerful feature of Backdrop that needs highlighting.

**Comment by docwilmot** [comment]
Of course @Graham-72. By all means edit away.

What I'm thinking also is the Quick Start Guide in the live version should have a bunch of links to the expanded versions of each topic.

**Comment by docwilmot** [comment]
@klonos I'm not sure how the tagging would work though since the docs are not being made via GitHub. And all the docs need screenshots, so there wont be any particular occasion where we'll need a specific "needs screenshots", unless we are creating issues for each document? I actually like that idea since we can have a way to alert that something changed or needs changing. But OTOH GDocs already allow inline commenting, and I think inline comments would be more efficient, so we'd be duplicating comments if we did issues too.

How are we going to work this?

**Comment by Graham-72** [comment]
>the live version should have a bunch of links

:+1:

**Comment by klonos** [comment]
@docwilmot that's why in the past I suggested working with unpublished revisions directly on backdropcms.org + comments so we can elaborate. The book nodes themselves could be tagged with a separate "documentation status" vocabulary. Pretty much like they do it in d.org:

![d org-documentation_page_status_tags](https://cloud.githubusercontent.com/assets/2423362/15632594/3a3b6280-25db-11e6-82b0-46729e9e519d.gif)

**Comment by klonos** [comment]
...unless we create a backdrop-docs repo in github and have separate .md files per documentation page and also use the folder structure as documentation sections. That would allow us to file issues and PRs against existing documentation + have comments available on them.

On the backdropcms.org side, we could have the documentation nodes have a link field (pointing to the respective github URL) instead of an actual body field.

Just a thought.

**Comment by jenlampton** [comment]
We tried docs on github, but its really a bad medium for managing text.
Every typo fix and wording change causes silly diffs. Revisions ++

**Comment by docwilmot** [comment]
@Graham-72 @jenlampton 
I've gotten a good bunch of written material now, would be grateful for some editing, commenting, nitpicking to begin. I've updated the first post to include the GDocs directory.

I plan to tackle guides for users, taxonomy, views, comments and themes next, not necessarily in this order.

There are no links yet of course because cant be sure what final URLs are.

**Comment by klonos** [comment]
> We tried docs on github, but its really a bad medium for managing text. Every typo fix and wording change causes silly diffs. Revisions ++

I have a really good feeling about [GitBook](https://www.gitbook.com) / https://github.com/backdrop-ops/backdropcms.org/issues/256. If we decide to use it, I should return here and go through this entire issue because we had some pretty good ideas and todos that would be a shame to get lost.

**Comment by docwilmot** [comment]
@klonos @jenlampton @Graham-72 I've created a checklist of docs on GDocs which are ready for some review (see first post). Theyre still rough, but enough to get some direction I hope. I'm not quite sure what is the best next step TBH. Options:
- paste all as content on BAckdropCMS.org now and edit/approve there.
- keep on Gdocs till approved
- use some other documentation collaboration tool

factors to consider:
- how to track comments and suggestions. Open an issue here for each doc and use the docs tags? Comments directly on the GDocs?
- how to manage images.

@jenlampton given the amount of back and forth so far, I think in this case it would be nice to have a final decision so we can move on with a plan. Maybe we can deliberate on a better plan for the maintenance of docs long term, but for now, would appreciate if you would feel free to lay down the law.

**Comment by jenlampton** [comment]
> paste all as content on BackdropCMS.org now and edit/approve there.

I'm in favor of this approach, and would also love to approve early and edit as much as necessary later. As long as nothing we say is absolutely wrong, I believe some info is better than no info! :)

**Comment by jenlampton** [comment]
> how to track comments and suggestions. Open an issue here for each doc and use the docs tags? 

Long term this might become cumbersome to manage the issue queue for minor docs changes.  Quick / easy ones (that don't require any discussion) can probably be made directly to the pages (as you see fit) without a GitHub issue. Let's not get too bogged down in the administration of the content, we need to keep you guys happy + healthy :)

**Comment by docwilmot** [comment]
Great. We'll go with that. I've saved them as unpublished, will continue to do so unless you think otherwise; I noticed the couple by @Graham-72 were published, but with no menu links that isn't really an issue I guess.
One final decision: how to organize on BackdropCMS. I can just save everything as a `documentation` content type for now, but we'll need to create a Book to put them in. Also, there's not much hierarchy now since theyre mostly just single long docs for each topic, but will need to decide on the organization, and the menu links, at some point. May be easier to decide on those now so new docs can just be put in the right location on creation, rather than having to move them around.

**Comment by jenlampton** [comment]
@docwilmot I like the idea of putting them into a book. Let me see if that's easy to jus turn on for you...

**Comment by jenlampton** [comment]
@docwilmot how's that, can you create a new book now? and add documentation items into it?

**Comment by docwilmot** [comment]
Yep, we're good now, thanks.

**Comment by klonos** [comment]
Great! I will start proofreading and adding screenshots as time permits. One thing: The book navigation is not that helpful the way we have it at the bottom of the page. I really liked (or I guess am used to) the way the documentation structure is presented on a right sidebar on d.org. Can we do something about that?

**Comment by docwilmot** [comment]
> I really liked (or I guess am used to) the way the documentation structure is presented on a right sidebar as an index on d.org. Can we do something about that?

Book module creates that block automatically. It just needs to be added to the layout.

**Comment by jenlampton** [comment]
Yep, good idea.

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by jenlampton** [comment]
Okay, I set up all Documentation pages to show the book navigation in the right sidebar:
https://backdropcms.org/node/706 is a good example.

**Comment by jenlampton** [comment]
@docwilmot it looks like some of the documentation is in docs content types, and some is in book. Is there a reason for that? (Is something missing from the documentation content type that needs to be added?) lmk! :)

**Comment by klonos** [comment]
Thanx @jenlampton 👍

**Comment by klonos** [comment]
...not much experience in using the Book module, but part of the index (what seems to be the level of the current page and below) is duplicated at the bottom of the page as a `.book-navigation` `<nav>`, after the body text and before the `.book-pager`. The redundancy is more prominent in book pages with short body text like https://backdropcms.org/node/468. Any way to remove that now that we have the full index in the sidebar?

Also, as a note/todo: we should set up automatic paths for books and book pages.

**Comment by jenlampton** [comment]
Another great point. I have created https://github.com/backdrop/backdrop-issues/issues/1976 to add a UI for book navigation display preferences, but short-term I can take it out in the theme layer.

**Comment by jenlampton** [comment]
I decided to leave the navigation (up / next / previous) but delete the tree. What do you think?
https://backdropcms.org/user-guide/backdrop-cms-user-guide

**Comment by jenlampton** [comment]
oops.

**Comment by docwilmot** [comment]
> @docwilmot it looks like some of the documentation is in documentation nodes, and some is in book nodes. Was that intentional?

Wasn't intentional. I forgot that the "Add child page" link defaults to using the Book content type. I used that link several times. Will manually transfer to documentation content type and delete the Book pages.

**Comment by jenlampton** [comment]
>  I forgot that the "Add child page" link defaults to using the Book content type. 

I'll see if I can fix that for you...

**Comment by docwilmot** [comment]
> I decided to leave the navigation (up / next / previous) but delete the tree. What do you think?

I think we should leave the tree in place as it was. Especially since our docs are likely to be long form, it may be best to have the tree at the end rather than scrolling back to the top to find it. Unless you can fix the position of the block. And also, certain people (me) are used to it right there.

**Comment by jenlampton** [comment]
Do we have a documentation-page on creating documentation?  I would like to take some notes as I am doing a review :) so far... I looooooooooove it!!!! :)

**Comment by jenlampton** [comment]
> What do you think?

haha, well of course there are three of us and we seem to all have 3 different opinions :) 
me: tree at right nav at bottom, 
@klonos: nothing at bottom, 
@docwilmot: everything at bottom.  
Maybe we should ask a larger audience of users? @cellear @Graham-72 @quicksketch @wesruv ?

**Comment by Graham-72** [comment]
I like what I am seeing now, with the book navigation tree in a right hand column and with forward,backward, up links at the bottom. Works well on my tablet too.

**Comment by klonos** [comment]
> I decided to leave the navigation (up / next / previous) but delete the tree. What do you think?

👍 

> @klonos: nothing at bottom...

...no, no. I like the `.book-pager` at the bottom. I never said anything about removing that.

**Comment by klonos** [comment]
...I think it would be nice if we could make the index at the right be sticky somehow. That would keep it within the users' view and they could scroll independently from the main documentation text.

**Comment by docwilmot** [comment]
Whats the consensus on the name of the Layout UI page? I had issues deciding which while writing the docs.
- Layout UI
- Layout settings page
- Layout edit/configure page
- Add-blocks page

Also how do you feel about grouping some of the definitional docs under a single group, e.g.: 
Quick start guide
- Modules themes and layouts
- Nodes and Entities

Or leave as is?

Also, would be good to have some way to report inaccurate docs, or to request new docs. Forums?

**Comment by klonos** [comment]
I think that the most "accurate" way to call things would be:

"Layouts list" for `/admin/structure/layouts`
"Manage layout blocks & regions" for `/admin/structure/layouts/manage/[layout_name]`
"Layout settings" for `/admin/structure/layouts/manage/default/settings`
"Layout**s** settings" for `/admin/structure/layouts/settings`

Now, this is to refer to things in documentation. But we need to be consistent in the UI too (I will file a separate issue against the backdrop repo for that).

Rationale:

`/admin/structure/layouts` currently has a tab with "List layouts" as label. That's OK and we could either use that in documentation, or reverse the order of words and be calling it "Layouts list". I don't think that there is much room for disambiguation there. It's just that in the UI (admin bar), we decided to have "List x" menu items for consistency.

`/admin/structure/layouts/manage/[layout_name]` currently has an "Edit layout" label. Now, I believe that "Place blocks page" is © by @jenlampton 😄 and that's what she likes calling it lately ..."Add blocks" doesn't fully describe the page because you can also move blocks around or remove them and you can also configure their settings or create custom blocks. "Layout UI" is generic enough to be describing all these tasks (and I personally use this), but then again perhaps it's generic to a point that some might find it very vague. What we are actually doing there is managing blocks (even the path has the word "manage" in it).

Now, there are things going on in that page that do not quite fit in the "manage blocks" realm...

- We have "Configure region" as secondary actions in the region dropbuttons.

- We have the "Page title type" dropdown menu. ...but there are plans to move that to a modal and have it pop up out of a "Configure" button in the "Page title" pseudo-region + most likely allow that to be moved around. So it will become more like a psuedo-block. (or an actual block, if we end up making the page title an actual field). 

Point is that using "edit" for that page was an easy way out of these dilemmas and so is using "Layout UI". On the other hand, "Manage layout blocks & regions" although accurate, is too long. Even if we omitted the "layout" part and called it "Manage blocks & regions", it's still long.

I left "Layout**s** settings" and "Layout settings" for last for the obvious reason that they will be easy to confuse the two. The fact that the second word ("settings") starts with an "s", which is the same letter that "Layouts" ends with, will make it "silent" when one talks fast. Any good ideas on that?

**Comment by jenlampton** [comment]
Related: we still don't know what to call these pages either:
https://github.com/backdrop/backdrop-issues/issues/1503

**Comment by docwilmot** [comment]
Starting a new topic at #263 to discuss the docs themselves as this discussion is getting a bit long at 105 comments.

**Comment by docwilmot** [comment]
> "Configure blocks" for /admin/structure/layouts/manage/[layout_name]

Calling it the "Configure blocks page" will probably also conflict visually with the "Block configuration form". Too many of the concepts are similar so I propose to not try to be too technically correct and call it something unique like "Layout designer" or "Layout Builder".

**Comment by wesruv** [comment]
Sticky tree on the side sounds appealing, can be tricky to implement though. Did a little bit of Googling, this seems like a good option, the "Scrollable Sticky Element" demo is particularly witty, and would be an issue we'd have.
http://leafo.net/sticky-kit/#examples

**Comment by docwilmot** [comment]
This continues on https://github.com/backdrop-ops/backdropcms.org/issues/263
